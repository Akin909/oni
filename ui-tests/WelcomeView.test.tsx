import { mount } from "enzyme"
import { Event } from "oni-types"
import * as React from "react"

import {
    WelcomeView,
    WelcomeViewProps,
    WelcomeViewState,
    IWelcomeInputEvent,
} from "./../browser/src/Editor/NeovimEditor/WelcomeBufferLayer"

describe("<WelcomeView />", () => {
    const buttons = [
        "button1",
        "button2",
        "button3",
        "button4",
        "button5",
        "button6",
        "button7",
        "button8",
    ]

    const commands = {
        openFile: "button1",
        openTutor: "button2",
        openDocs: "button3",
        openConfig: "button4",
        openThemes: "button5",
        openWorkspaceFolder: "button6",
        commandPalette: "button7",
        commandline: "button8",
    }

    const executeCommand = jest.fn()
    const inputEvent = new Event<IWelcomeInputEvent>("handleInputTestEvent")
    let handleInputSpy: jest.SpyInstance<WelcomeView["handleInput"]>

    afterEach(() => {
        if (handleInputSpy) {
            handleInputSpy.mockClear()
        }
        instance.setState({ selectedId: "button1", currentIndex: 0 })
    })

    const wrapper = mount<WelcomeViewProps, WelcomeViewState>(
        <WelcomeView
            active
            buttonIds={buttons}
            inputEvent={inputEvent}
            commands={commands}
            executeCommand={executeCommand}
        />,
    )
    const instance = wrapper.instance() as WelcomeView

    it("Should render without crashing", () => {
        expect(wrapper.length).toBe(1)
    })

    it("should default initial state to the first button id", () => {
        expect(instance.state.selectedId).toBe("button1")
    })

    it("should correctly update selection based on input", () => {
        instance.handleInput({ direction: 1, select: false })
        expect(instance.state.selectedId).toBe("button2")
    })

    it("should loop back to button if user navigates upwards at the first button", () => {
        instance.handleInput({ direction: -1, select: false })
        expect(instance.state.currentIndex).toBe(7)
        expect(instance.state.selectedId).toBe("button8")
    })

    it("should loop back to button if user navigates downwards at the last button", () => {
        instance.setState({ currentIndex: 7, selectedId: "button8" })
        instance.handleInput({ direction: 1, select: false })
        expect(instance.state.currentIndex).toBe(0)
        expect(instance.state.selectedId).toBe("button1")
    })
})
