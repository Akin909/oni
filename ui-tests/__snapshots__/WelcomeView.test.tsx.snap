// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomeView /> should match its last snapshot 1`] = `
<styled.div
  data-id="welcome-screen"
  height="100%"
  innerRef={
    Object {
      "current": null,
    }
  }
>
  <styled.div
    extension={
      Array [
        "
    width: 100%;
    padding-top: 32px;
    animation: ",
        "ckrRMN",
        " 0.25s ease-in 0.25s forwards};
",
      ]
    }
  >
    <styled.div />
    <styled.div
      alignment="flex-end"
    >
      <styled.div>
        Oni
      </styled.div>
      <styled.div>
        Modern Modal Editing
      </styled.div>
    </styled.div>
    <styled.div
      flex="0 0"
    >
      <styled.img
        src="images/oni-icon-no-border.svg"
      />
    </styled.div>
    <styled.div
      alignment="flex-start"
    >
      <div>
        https://onivim.io
      </div>
    </styled.div>
    <styled.div />
  </styled.div>
  <styled.div
    extension={
      Array [
        "
    width: 70%;
    height: 60%;
    box-sizing: border-box;
    padding: 0 1rem;
    margin-top: 64px;
    opacity: 1;
    border: 1px solid ",
        [Function],
        ";
    border-radius: 4px;
    justify-content: space-between;
    overflow: hidden;
    background-color: ",
        [Function],
        ";
    ",
        "
    box-shadow: inset 0 4px 8px 2px rgba(0, 0, 0, 0.2);
",
        ";
",
      ]
    }
  >
    <WelcomeCommandsView
      commands={
        Object {
          "commandPalette": Object {
            "command": "button7",
          },
          "commandline": Object {
            "command": "button8",
          },
          "openConfig": Object {
            "command": "button4",
          },
          "openDocs": Object {
            "command": "button3",
          },
          "openFile": Object {
            "command": "button1",
          },
          "openThemes": Object {
            "command": "button5",
          },
          "openTutor": Object {
            "command": "button2",
          },
          "openWorkspaceFolder": Object {
            "command": "button6",
          },
        }
      }
      executeCommand={[MockFunction]}
      selectedId="button1"
    />
    <styled.div>
      <styled.ul>
        <styled.div>
          Sessions
        </styled.div>
        <styled.li
          isSelected={false}
          key="test-1"
          onClick={[Function]}
        >
          <Icon
            name="file"
          />
           
          test
        </styled.li>
      </styled.ul>
    </styled.div>
  </styled.div>
</styled.div>
`;
